diff --git a/t/sni.t b/t/sni.t
index 9c8202a..c755d17 100644
--- a/t/sni.t
+++ b/t/sni.t
@@ -72,6 +72,8 @@ if ( $pid == 0 ) {
 	    print "not ok # client ssl connect $host - $SSL_ERROR\n";
 	    print "ok # skip connect failed\n";
 	}
+	# wait for server to send something to make sure finished accept
+	<$client>;
     }
     exit;
 }
@@ -83,6 +85,7 @@ for my $host (@tests) {
 	my $name = $csock->get_servername;
 	print "not " if ! $name or $name ne $host;
 	print "ok # server got SNI name $host\n";
+	print $csock "hi\n";
     } else {
 	print "not ok # server accept - $SSL_ERROR\n";
 	print "ok # skip accept failed\n";
