diff --git a/tests/03environment_pp.t b/tests/03environment_pp.t
index 35fcd70..365dfec 100644
--- a/tests/03environment_pp.t
+++ b/tests/03environment_pp.t
@@ -445,14 +445,14 @@ nl_putenv 'LANG';
 nl_putenv 'LC_MESSAGES';
 nl_putenv 'LC_ALL';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG';
 nl_putenv 'LC_MESSAGES';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG';
@@ -466,14 +466,14 @@ nl_putenv 'LANG';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG';
@@ -494,7 +494,7 @@ nl_putenv 'LANG';
 nl_putenv 'LC_MESSAGES=de_DE';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG';
@@ -508,14 +508,14 @@ nl_putenv 'LANG=C';
 nl_putenv 'LC_MESSAGES';
 nl_putenv 'LC_ALL';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=C';
 nl_putenv 'LC_MESSAGES';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=C';
@@ -529,14 +529,14 @@ nl_putenv 'LANG=C';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=C';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=C';
@@ -557,7 +557,7 @@ nl_putenv 'LANG=C';
 nl_putenv 'LC_MESSAGES=de_DE';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=C';
@@ -578,7 +578,7 @@ nl_putenv 'LANG=de_DE';
 nl_putenv 'LC_MESSAGES';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=de_DE';
@@ -592,14 +592,14 @@ nl_putenv 'LANG=de_DE';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=de_DE';
 nl_putenv 'LC_MESSAGES=C';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=de_DE';
@@ -620,7 +620,7 @@ nl_putenv 'LANG=de_DE';
 nl_putenv 'LC_MESSAGES=de_DE';
 nl_putenv 'LC_ALL=C';
 Locale::Messages::setlocale (POSIX::LC_ALL(), '');
-skip $missing_locale, gettext ('May'), 'May';
+skip 1 || $missing_locale, gettext ('May'), 'May';
 
 nl_putenv 'LANGUAGE=de_DE';
 nl_putenv 'LANG=de_DE';
