diff --git a/Makefile.PL b/Makefile.PL
index d00060b..4d5babd 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -3,6 +3,25 @@ use strict;
 use warnings;
 use ExtUtils::MakeMaker::CPANfile;
 
+my $test_requires; # Decide between TEST_REQUIRES / BUILD_REQUIRES / PREREQ_PM
+if ($ExtUtils::MakeMaker::VERSION >= 6.6303) {
+	#
+	# Good, we can use TEST_REQUIRES
+	#
+	$test_requires = 'TEST_REQUIRES';
+} elsif ($ExtUtils::MakeMaker::VERSION >= 6.5503) {
+	#
+	# Not new enough - use BUILD_REQUIRES
+	#
+	$test_requires = 'BUILD_REQUIRES';
+} else {
+	#
+	# Tool old version - use PREREQ_PM
+	#
+	$test_requires = 'PREREQ_PM';
+}
+my %test_requires = ( $test_requires => { 'Test::File' => 1.993 } );
+
 WriteMakefile(
   NAME => 'Module::CPANTS::Analyse',
   AUTHOR => [
@@ -23,6 +42,7 @@ WriteMakefile(
       'CPANTS','kwalitee',
     ],
   },
+  %test_requires,
 
   test => {TESTS => 't/*.t t/*/*.t'},
 );
diff --git a/t/analyse/manifest.t b/t/analyse/manifest.t
index 5aef039..2f542b1 100644
--- a/t/analyse/manifest.t
+++ b/t/analyse/manifest.t
@@ -1,6 +1,7 @@
 use strict;
 use warnings;
 use FindBin;
+use Test::File 1.993;
 use lib "$FindBin::Bin/../lib";
 use Module::CPANTS::TestAnalyse;
 
@@ -50,6 +51,10 @@ test_distribution {
 MANIFEST
 EOF
 
+  if (! Test::File->has_symlinks) {
+    diag "No symlink in this session";
+    return;
+  }
   eval { symlink "$dir/MANIFEST", "$dir/MANIFEST.lnk" };
   if ($@) {
     diag "symlink is not supported";
@@ -66,6 +71,10 @@ test_distribution {
 MANIFEST
 EOF
 
+  if (! Test::File->has_symlinks) {
+    diag "No symlink in this session";
+    return;
+  }
   eval { symlink "$dir/MANIFEST", "$dir/MANIFEST.lnk" };
   if ($@) {
     diag "symlink is not supported";
