diff --git a/t/01-fail-if-SYNOPSIS-has-errors.t b/t/01-fail-if-SYNOPSIS-has-errors.t
deleted file mode 100644
index 2d6eec4..0000000
--- a/t/01-fail-if-SYNOPSIS-has-errors.t
+++ /dev/null
@@ -1,19 +0,0 @@
-use Test::Builder::Tester tests => 1;
-use Test::More;
-use Test::Synopsis;
-
-# Test whether we indeed can detect errors in SYNOPSIS codes
-
-test_out('not ok 1 - t/lib/BrokenSYNOPSIS01.pm');
-test_diag(q{  Failed test 't/lib/BrokenSYNOPSIS01.pm'},
-    q{  at t/01-fail-if-SYNOPSIS-has-errors.t line } . line_num(+9) . q{.},
-    q{Global symbol "$x" requires explicit package name}
-     . ( ($^V and $^V gt 5.21.3)
-            ? ' (did you forget to declare "my $x"?)'
-            : ''
-        )
-     . q{ at t/lib/BrokenSYNOPSIS01.pm line 18.},
-);
-synopsis_ok("t/lib/BrokenSYNOPSIS01.pm");
-test_test("synopsis fail works");
-
diff --git a/t/02-END-token-errors.t b/t/02-END-token-errors.t
deleted file mode 100644
index 6c7a089..0000000
--- a/t/02-END-token-errors.t
+++ /dev/null
@@ -1,18 +0,0 @@
-use Test::Builder::Tester tests => 1;
-use Test::More;
-use Test::Synopsis;
-
-# Test whether we indeed can detect errors in SYNOPSIS codes
-
-test_out('not ok 1 - t/lib/ENDInPodWithError.pm');
-test_diag(q{  Failed test 't/lib/ENDInPodWithError.pm'},
-    q{  at t/02-END-token-errors.t line } . line_num(+9) . q{.},
-    q{Global symbol "$x" requires explicit package name}
-    . ( ($^V and $^V gt 5.21.3)
-        ? ' (did you forget to declare "my $x"?)'
-        : ''
-    )
-    . q{ at t/lib/ENDInPodWithError.pm line 24.},
-);
-synopsis_ok("t/lib/ENDInPodWithError.pm");
-test_test("synopsis fail works");
diff --git a/t/05-multi-chunks-clash.t b/t/05-multi-chunks-clash.t
deleted file mode 100644
index cb18f5f..0000000
--- a/t/05-multi-chunks-clash.t
+++ /dev/null
@@ -1,13 +0,0 @@
-use Test::Builder::Tester tests => 1;
-use Test::More;
-use Test::Synopsis;
-
-test_out('ok 1 - t/lib/TestMultipleChunks.pm (section 1)
-not ok 2 - t/lib/TestMultipleChunks.pm (section 2)');
-
-test_diag(q{  Failed test 't/lib/TestMultipleChunks.pm (section 2)'
-#   at t/05-multi-chunks-clash.t line 12.
-# Bareword "bob" not allowed while "strict subs" in use at t/lib/TestMultipleChunks.pm line 29.});
-
-synopsis_ok("t/lib/TestMultipleChunks.pm");
-test_test("synopsis with multiple chunks fails");
diff --git a/t/12-DATA-token-errors.t b/t/12-DATA-token-errors.t
deleted file mode 100644
index 91323f7..0000000
--- a/t/12-DATA-token-errors.t
+++ /dev/null
@@ -1,18 +0,0 @@
-use Test::Builder::Tester tests => 1;
-use Test::More;
-use Test::Synopsis;
-
-# Test whether we indeed can detect errors in SYNOPSIS codes
-
-test_out('not ok 1 - t/lib/Test12DATAInPodWithError.pm');
-test_diag(q{  Failed test 't/lib/Test12DATAInPodWithError.pm'},
-    q{  at t/12-DATA-token-errors.t line } . line_num(+9) . q{.},
-    q{Global symbol "$x" requires explicit package name}
-    . ( ($^V and $^V gt 5.21.3)
-        ? ' (did you forget to declare "my $x"?)'
-        : ''
-    )
-    . q{ at t/lib/Test12DATAInPodWithError.pm line 24.},
-);
-synopsis_ok("t/lib/Test12DATAInPodWithError.pm");
-test_test("synopsis fail works");
