diff -Naur Win32-NetAdmin-0.13.old/t/netadmin.t Win32-NetAdmin-0.13.new/t/netadmin.t
--- Win32-NetAdmin-0.13.old/t/netadmin.t	2013-12-10 02:07:52.000000000 +0100
+++ Win32-NetAdmin-0.13.new/t/netadmin.t	2022-12-02 05:47:59.264499100 +0100
@@ -21,7 +21,7 @@
 
 my $serverName   = '';
 my $userName     = 'TestUser';
-my $password     = 'password';
+my $password     = 'pa55w0r&';
 my $passwordAge  = 0;
 my $privilege    = USER_PRIV_USER;
 my $homeDir      = 'c:\\';
