diff --git a/cmake/LZMACodec.cmake b/cmake/LZMACodec.cmake
index 06b14c1e..7bafbdfc 100644
--- a/cmake/LZMACodec.cmake
+++ b/cmake/LZMACodec.cmake
@@ -27,7 +27,7 @@
 # liblzma2
 set(LZMA_SUPPORT FALSE)
 find_package(LibLZMA)
-option(lzma "use liblzma (required for LZMA2 compression)" ${LZMA_FOUND})
+option(lzma "use liblzma (required for LZMA2 compression)" ${LIBLZMA_FOUND})
 if (lzma AND LIBLZMA_FOUND)
     set(LZMA_SUPPORT TRUE)
 endif()
